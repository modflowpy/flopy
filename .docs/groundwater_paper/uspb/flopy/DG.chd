# CHD for MODFLOW, generated by Flopy.
       140
       140         0 # stress period 0
         5        69       179 1165.720703 1165.720703
         5        68       179 1167.066895 1167.066895
         5        68       178 1168.655518 1168.655518
         5        67       178 1170.057495 1170.057495
         5        67       177 1171.646240 1171.646240
         5        66       177 1173.048218 1173.048218
         5        66       176 1174.636963 1174.636963
         5        65       176 1176.038940 1176.038940
         5        65       175 1177.506226 1177.506226
         5        65       174 1179.029541 1179.029541
         5        64       174 1180.431519 1180.431519
         5        64       173 1182.020264 1182.020264
         5        63       173 1183.422241 1183.422241
         5        63       172 1185.010986 1185.010986
         5        62       172 1186.412964 1186.412964
         5        62       171 1188.001587 1188.001587
         5        61       171 1189.403564 1189.403564
         5        61       170 1190.562134 1190.562134
         4        37       235 1245.777100 1245.777100
         5        37       235 1245.777100 1245.777100
         4        37       234 1245.328125 1245.328125
         5        37       234 1245.328125 1245.328125
         4        38       234 1244.399170 1244.399170
         5        38       234 1244.399170 1244.399170
         4        38       233 1243.276733 1243.276733
         5        38       233 1243.276733 1243.276733
         4        39       233 1242.347900 1242.347900
         5        39       233 1242.347900 1242.347900
         4        39       232 1241.225342 1241.225342
         5        39       232 1241.225342 1241.225342
         4        40       232 1240.296509 1240.296509
         5        40       232 1240.296509 1240.296509
         4        40       231 1239.173950 1239.173950
         5        40       231 1239.173950 1239.173950
         4        41       231 1238.245117 1238.245117
         5        41       231 1238.245117 1238.245117
         4        41       230 1237.194580 1237.194580
         5        41       230 1237.194580 1237.194580
         4        41       229 1236.193726 1236.193726
         5        41       229 1236.193726 1236.193726
         4        42       229 1235.264893 1235.264893
         5        42       229 1235.264893 1235.264893
         4        42       228 1234.142334 1234.142334
         5        42       228 1234.142334 1234.142334
         4        43       228 1233.213501 1233.213501
         5        43       228 1233.213501 1233.213501
         4        43       227 1232.091064 1232.091064
         5        43       227 1232.091064 1232.091064
         4        44       227 1231.162231 1231.162231
         5        44       227 1231.162231 1231.162231
         4        44       226 1230.039673 1230.039673
         5        44       226 1230.039673 1230.039673
         4        45       226 1229.110840 1229.110840
         5        45       226 1229.110840 1229.110840
         4        45       225 1227.988281 1227.988281
         5        45       225 1227.988281 1227.988281
         4        46       225 1227.059448 1227.059448
         5        46       225 1227.059448 1227.059448
         4        46       224 1226.048462 1226.048462
         5        46       224 1226.048462 1226.048462
         4        46       223 1225.008057 1225.008057
         5        46       223 1225.008057 1225.008057
         4        47       223 1224.079224 1224.079224
         5        47       223 1224.079224 1224.079224
         4        47       222 1222.956665 1222.956665
         5        47       222 1222.956665 1222.956665
         4        48       222 1222.027832 1222.027832
         5        48       222 1222.027832 1222.027832
         4        48       221 1220.905396 1220.905396
         5        48       221 1220.905396 1220.905396
         4        49       221 1219.976440 1219.976440
         5        49       221 1219.976440 1219.976440
         4        49       220 1218.854004 1218.854004
         5        49       220 1218.854004 1218.854004
         4        50       220 1217.925171 1217.925171
         5        50       220 1217.925171 1217.925171
         4        50       219 1216.802612 1216.802612
         5        50       219 1216.802612 1216.802612
         4        51       219 1215.873779 1215.873779
         5        51       219 1215.873779 1215.873779
         4        51       218 1214.902344 1214.902344
         5        51       218 1214.902344 1214.902344
         4        51       217 1213.822388 1213.822388
         5        51       217 1213.822388 1213.822388
         4        52       217 1212.893555 1212.893555
         5        52       217 1212.893555 1212.893555
         4        52       216 1211.770996 1211.770996
         5        52       216 1211.770996 1211.770996
         4        53       216 1210.842163 1210.842163
         5        53       216 1210.842163 1210.842163
         4        53       215 1209.719604 1209.719604
         5        53       215 1209.719604 1209.719604
         4        54       215 1208.790771 1208.790771
         5        54       215 1208.790771 1208.790771
         4        54       214 1207.668335 1207.668335
         5        54       214 1207.668335 1207.668335
         4        55       214 1206.739502 1206.739502
         5        55       214 1206.739502 1206.739502
         4        55       213 1205.616943 1205.616943
         5        55       213 1205.616943 1205.616943
         4        56       213 1204.688110 1204.688110
         5        56       213 1204.688110 1204.688110
         4        56       212 1203.756348 1203.756348
         5        56       212 1203.756348 1203.756348
         4        56       211 1202.636719 1202.636719
         5        56       211 1202.636719 1202.636719
         4        57       211 1201.707886 1201.707886
         5        57       211 1201.707886 1201.707886
         4        57       210 1200.585327 1200.585327
         5        57       210 1200.585327 1200.585327
         4        58       210 1199.656494 1199.656494
         5        58       210 1199.656494 1199.656494
         4        58       209 1198.533936 1198.533936
         5        58       209 1198.533936 1198.533936
         4        59       209 1197.605103 1197.605103
         5        59       209 1197.605103 1197.605103
         4        59       208 1196.482666 1196.482666
         5        59       208 1196.482666 1196.482666
         4        60       208 1195.553711 1195.553711
         5        60       208 1195.553711 1195.553711
         4        60       207 1194.467896 1194.467896
         5        60       207 1194.467896 1194.467896
         4        60       206 1193.502441 1193.502441
         5        60       206 1193.502441 1193.502441
         4        61       206 1192.573608 1192.573608
         5        61       206 1192.573608 1192.573608
         4        61       205 1191.451050 1191.451050
         5        61       205 1191.451050 1191.451050
         4        62       205 1190.522217 1190.522217
         5        62       205 1190.522217 1190.522217
         4        62       204 1189.399658 1189.399658
         5        62       204 1189.399658 1189.399658
         4        63       204 1188.470825 1188.470825
         5        63       204 1188.470825 1188.470825
         4        63       203 1187.483032 1187.483032
         5        63       203 1187.483032 1187.483032
         5        70       186 1158.099976 1158.099976
         5        70       185 1158.099976 1158.099976
         5        71       185 1158.099976 1158.099976
         5        71       184 1158.099976 1158.099976
       140         0 # stress period 1
         5        69       179 1165.720703 1165.720703
         5        68       179 1167.066895 1167.066895
         5        68       178 1168.655518 1168.655518
         5        67       178 1170.057495 1170.057495
         5        67       177 1171.646240 1171.646240
         5        66       177 1173.048218 1173.048218
         5        66       176 1174.636963 1174.636963
         5        65       176 1176.038940 1176.038940
         5        65       175 1177.506226 1177.506226
         5        65       174 1179.029541 1179.029541
         5        64       174 1180.431519 1180.431519
         5        64       173 1182.020264 1182.020264
         5        63       173 1183.422241 1183.422241
         5        63       172 1185.010986 1185.010986
         5        62       172 1186.412964 1186.412964
         5        62       171 1188.001587 1188.001587
         5        61       171 1189.403564 1189.403564
         5        61       170 1190.562134 1190.562134
         4        37       235 1245.777100 1245.777100
         5        37       235 1245.777100 1245.777100
         4        37       234 1245.328125 1245.328125
         5        37       234 1245.328125 1245.328125
         4        38       234 1244.399170 1244.399170
         5        38       234 1244.399170 1244.399170
         4        38       233 1243.276733 1243.276733
         5        38       233 1243.276733 1243.276733
         4        39       233 1242.347900 1242.347900
         5        39       233 1242.347900 1242.347900
         4        39       232 1241.225342 1241.225342
         5        39       232 1241.225342 1241.225342
         4        40       232 1240.296509 1240.296509
         5        40       232 1240.296509 1240.296509
         4        40       231 1239.173950 1239.173950
         5        40       231 1239.173950 1239.173950
         4        41       231 1238.245117 1238.245117
         5        41       231 1238.245117 1238.245117
         4        41       230 1237.194580 1237.194580
         5        41       230 1237.194580 1237.194580
         4        41       229 1236.193726 1236.193726
         5        41       229 1236.193726 1236.193726
         4        42       229 1235.264893 1235.264893
         5        42       229 1235.264893 1235.264893
         4        42       228 1234.142334 1234.142334
         5        42       228 1234.142334 1234.142334
         4        43       228 1233.213501 1233.213501
         5        43       228 1233.213501 1233.213501
         4        43       227 1232.091064 1232.091064
         5        43       227 1232.091064 1232.091064
         4        44       227 1231.162231 1231.162231
         5        44       227 1231.162231 1231.162231
         4        44       226 1230.039673 1230.039673
         5        44       226 1230.039673 1230.039673
         4        45       226 1229.110840 1229.110840
         5        45       226 1229.110840 1229.110840
         4        45       225 1227.988281 1227.988281
         5        45       225 1227.988281 1227.988281
         4        46       225 1227.059448 1227.059448
         5        46       225 1227.059448 1227.059448
         4        46       224 1226.048462 1226.048462
         5        46       224 1226.048462 1226.048462
         4        46       223 1225.008057 1225.008057
         5        46       223 1225.008057 1225.008057
         4        47       223 1224.079224 1224.079224
         5        47       223 1224.079224 1224.079224
         4        47       222 1222.956665 1222.956665
         5        47       222 1222.956665 1222.956665
         4        48       222 1222.027832 1222.027832
         5        48       222 1222.027832 1222.027832
         4        48       221 1220.905396 1220.905396
         5        48       221 1220.905396 1220.905396
         4        49       221 1219.976440 1219.976440
         5        49       221 1219.976440 1219.976440
         4        49       220 1218.854004 1218.854004
         5        49       220 1218.854004 1218.854004
         4        50       220 1217.925171 1217.925171
         5        50       220 1217.925171 1217.925171
         4        50       219 1216.802612 1216.802612
         5        50       219 1216.802612 1216.802612
         4        51       219 1215.873779 1215.873779
         5        51       219 1215.873779 1215.873779
         4        51       218 1214.902344 1214.902344
         5        51       218 1214.902344 1214.902344
         4        51       217 1213.822388 1213.822388
         5        51       217 1213.822388 1213.822388
         4        52       217 1212.893555 1212.893555
         5        52       217 1212.893555 1212.893555
         4        52       216 1211.770996 1211.770996
         5        52       216 1211.770996 1211.770996
         4        53       216 1210.842163 1210.842163
         5        53       216 1210.842163 1210.842163
         4        53       215 1209.719604 1209.719604
         5        53       215 1209.719604 1209.719604
         4        54       215 1208.790771 1208.790771
         5        54       215 1208.790771 1208.790771
         4        54       214 1207.668335 1207.668335
         5        54       214 1207.668335 1207.668335
         4        55       214 1206.739502 1206.739502
         5        55       214 1206.739502 1206.739502
         4        55       213 1205.616943 1205.616943
         5        55       213 1205.616943 1205.616943
         4        56       213 1204.688110 1204.688110
         5        56       213 1204.688110 1204.688110
         4        56       212 1203.756348 1203.756348
         5        56       212 1203.756348 1203.756348
         4        56       211 1202.636719 1202.636719
         5        56       211 1202.636719 1202.636719
         4        57       211 1201.707886 1201.707886
         5        57       211 1201.707886 1201.707886
         4        57       210 1200.585327 1200.585327
         5        57       210 1200.585327 1200.585327
         4        58       210 1199.656494 1199.656494
         5        58       210 1199.656494 1199.656494
         4        58       209 1198.533936 1198.533936
         5        58       209 1198.533936 1198.533936
         4        59       209 1197.605103 1197.605103
         5        59       209 1197.605103 1197.605103
         4        59       208 1196.482666 1196.482666
         5        59       208 1196.482666 1196.482666
         4        60       208 1195.553711 1195.553711
         5        60       208 1195.553711 1195.553711
         4        60       207 1194.467896 1194.467896
         5        60       207 1194.467896 1194.467896
         4        60       206 1193.502441 1193.502441
         5        60       206 1193.502441 1193.502441
         4        61       206 1192.573608 1192.573608
         5        61       206 1192.573608 1192.573608
         4        61       205 1191.451050 1191.451050
         5        61       205 1191.451050 1191.451050
         4        62       205 1190.522217 1190.522217
         5        62       205 1190.522217 1190.522217
         4        62       204 1189.399658 1189.399658
         5        62       204 1189.399658 1189.399658
         4        63       204 1188.470825 1188.470825
         5        63       204 1188.470825 1188.470825
         4        63       203 1187.483032 1187.483032
         5        63       203 1187.483032 1187.483032
         5        70       186 1158.099976 1158.099976
         5        70       185 1158.099976 1158.099976
         5        71       185 1158.099976 1158.099976
         5        71       184 1158.099976 1158.099976
