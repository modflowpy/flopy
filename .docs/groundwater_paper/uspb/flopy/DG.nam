# Name file for mf2005, generated by Flopy.
LIST           2 DG.list
DIS           11 DG.dis
BAS6          13 DG.bas
LPF           15 DG.lpf
DRN           21 DG.drn
EVT           22 DG.evt
CHD           24 DG.chd
PCG           27 DG.pcg
OC            14 DG.oc
DATA(BINARY)  51 DG.hds REPLACE
DATA(BINARY)  52 DG.ddn REPLACE
DATA(BINARY)  53 DG.cbc REPLACE
RCH           19 DG.rch
STR          118 DG.str
WEL           20 DG.wel
