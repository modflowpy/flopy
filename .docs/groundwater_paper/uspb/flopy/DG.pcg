# PCG for MODFLOW, generated by Flopy.
39 50 1 0
0.0001 0.1 1.0 0 0 0 1.0
