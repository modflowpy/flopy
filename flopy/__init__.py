"""
The FloPy package consists of a set of Python scripts to run MODFLOW, MT3D,
SEAWAT and other MODFLOW-related groundwater programs. FloPy enables you to
run all these programs with Python scripts. The FloPy project started in 2009
and has grown to a fairly complete set of scripts with a growing user base.

This version of Flopy (FloPy3) was released in December 2015 with a few great
enhancements that make FloPy3 backwards incompatible. The first significant
change is that FloPy3 uses zero-based indexing everywhere, which means that
all layers, rows, columns, and stress periods start numbering at zero. This
change was made for consistency as all array-indexing was already zero-based
(as are all arrays in Python). This may take a little getting-used-to, but
hopefully will avoid confusion in the future. A second significant enhancement
concerns the ability to specify time-varying boundary conditions that are
specified with a sequence of layer-row-column-values, like the WEL and GHB
packages. A variety of flexible and readable ways have been implemented to
specify these boundary conditions. FloPy is an open-source project and any
assistance is welcomed. Please email the development team if you want to
contribute.

"""

from .version import __version__, __author__, __author_email__

# imports
from . import modflow
from . import mt3d
from . import seawat
from . import modpath
from . import modflowlgr
from . import utils
from . import plot
from . import export
from . import pest
from . import mf6
from . import discretization

from .mbase import run_model, which
