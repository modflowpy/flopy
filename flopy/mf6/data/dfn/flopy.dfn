name ts_filerecord
construct_package ts
construct_data timeseries
parameter_name timeseries

name obs_filerecord
construct_package obs
construct_data continuous
parameter_name observations

name tas_filerecord
construct_package tas
construct_data tas_array
parameter_name timearrayseries