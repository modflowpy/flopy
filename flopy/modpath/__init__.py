from .mp6 import Modpath6
from .mp6bas import Modpath6Bas
from .mp6sim import Modpath6Sim
from .mp7 import Modpath7
from .mp7bas import Modpath7Bas
from .mp7sim import Modpath7Sim
from .mp7particlegroup import (
    ParticleGroup,
    ParticleGroupLRCTemplate,
    ParticleGroupNodeTemplate,
)
from .mp7particledata import (
    ParticleData,
    FaceDataType,
    CellDataType,
    LRCParticleData,
    NodeParticleData,
)
