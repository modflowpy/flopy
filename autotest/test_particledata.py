import numpy as np

from flopy.modpath import ParticleData

structured_plocs = [(1, 1, 1), (1, 1, 2)]
structured_dtype = np.dtype(
    [
        ("k", "<i4"),
        ("i", "<i4"),
        ("j", "<i4"),
        ("localx", "<f4"),
        ("localy", "<f4"),
        ("localz", "<f4"),
        ("timeoffset", "<f4"),
        ("drape", "<i4"),
    ]
)
structured_array = np.core.records.fromrecords(
    [
        (1, 1, 1, 0.5, 0.5, 0.5, 0.0, 0),
        (1, 1, 2, 0.5, 0.5, 0.5, 0.0, 0),
    ],
    dtype=structured_dtype,
)


def test_particledata_structured_partlocs_as_list_of_tuples():
    locs = structured_plocs
    data = ParticleData(partlocs=locs, structured=True)

    assert data.particlecount == 2
    assert data.dtype == structured_dtype
    assert np.array_equal(data.particledata, structured_array)


def test_particledata_structured_partlocs_as_ndarray():
    locs = np.array(structured_plocs)
    data = ParticleData(partlocs=locs, structured=True)

    assert data.particlecount == 2
    assert data.dtype == structured_dtype
    assert np.array_equal(data.particledata, structured_array)


def test_particledata_structured_partlocs_as_list_of_lists():
    locs = [list(p) for p in structured_plocs]
    data = ParticleData(partlocs=locs, structured=True)

    assert data.particlecount == 2
    assert data.dtype == structured_dtype
    assert np.array_equal(data.particledata, structured_array)
