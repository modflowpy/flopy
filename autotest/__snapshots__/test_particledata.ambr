# serializer version: 1
# name: test_lrcparticledata_to_prp_1_per_face
  list([
    list([
      0,
      0,
      1,
      1,
      1.0,
      0.5,
      7.5,
    ]),
    list([
      1,
      0,
      1,
      1,
      2.0,
      0.5,
      7.5,
    ]),
    list([
      2,
      0,
      1,
      1,
      1.5,
      0.0,
      7.5,
    ]),
    list([
      3,
      0,
      1,
      1,
      1.5,
      1.0,
      7.5,
    ]),
    list([
      4,
      0,
      1,
      1,
      1.5,
      0.5,
      5.0,
    ]),
    list([
      5,
      0,
      1,
      1,
      1.5,
      0.5,
      10.0,
    ]),
  ])
# ---
# name: test_lrcparticledata_to_prp_divisions_defaults[False]
  list([
    list([
      0,
      0,
      0,
      1,
      1.1666666666666667,
      1.1666666666666667,
      5.833333333333333,
    ]),
    list([
      1,
      0,
      0,
      1,
      1.5,
      1.1666666666666667,
      5.833333333333333,
    ]),
    list([
      2,
      0,
      0,
      1,
      1.8333333333333335,
      1.1666666666666667,
      5.833333333333333,
    ]),
    list([
      3,
      0,
      0,
      1,
      1.1666666666666667,
      1.5,
      5.833333333333333,
    ]),
    list([
      4,
      0,
      0,
      1,
      1.5,
      1.5,
      5.833333333333333,
    ]),
    list([
      5,
      0,
      0,
      1,
      1.8333333333333335,
      1.5,
      5.833333333333333,
    ]),
    list([
      6,
      0,
      0,
      1,
      1.1666666666666667,
      1.8333333333333335,
      5.833333333333333,
    ]),
    list([
      7,
      0,
      0,
      1,
      1.5,
      1.8333333333333335,
      5.833333333333333,
    ]),
    list([
      8,
      0,
      0,
      1,
      1.8333333333333335,
      1.8333333333333335,
      5.833333333333333,
    ]),
    list([
      9,
      0,
      0,
      1,
      1.1666666666666667,
      1.1666666666666667,
      7.5,
    ]),
    list([
      10,
      0,
      0,
      1,
      1.5,
      1.1666666666666667,
      7.5,
    ]),
    list([
      11,
      0,
      0,
      1,
      1.8333333333333335,
      1.1666666666666667,
      7.5,
    ]),
    list([
      12,
      0,
      0,
      1,
      1.1666666666666667,
      1.5,
      7.5,
    ]),
    list([
      13,
      0,
      0,
      1,
      1.5,
      1.5,
      7.5,
    ]),
    list([
      14,
      0,
      0,
      1,
      1.8333333333333335,
      1.5,
      7.5,
    ]),
    list([
      15,
      0,
      0,
      1,
      1.1666666666666667,
      1.8333333333333335,
      7.5,
    ]),
    list([
      16,
      0,
      0,
      1,
      1.5,
      1.8333333333333335,
      7.5,
    ]),
    list([
      17,
      0,
      0,
      1,
      1.8333333333333335,
      1.8333333333333335,
      7.5,
    ]),
    list([
      18,
      0,
      0,
      1,
      1.1666666666666667,
      1.1666666666666667,
      9.166666666666666,
    ]),
    list([
      19,
      0,
      0,
      1,
      1.5,
      1.1666666666666667,
      9.166666666666666,
    ]),
    list([
      20,
      0,
      0,
      1,
      1.8333333333333335,
      1.1666666666666667,
      9.166666666666666,
    ]),
    list([
      21,
      0,
      0,
      1,
      1.1666666666666667,
      1.5,
      9.166666666666666,
    ]),
    list([
      22,
      0,
      0,
      1,
      1.5,
      1.5,
      9.166666666666666,
    ]),
    list([
      23,
      0,
      0,
      1,
      1.8333333333333335,
      1.5,
      9.166666666666666,
    ]),
    list([
      24,
      0,
      0,
      1,
      1.1666666666666667,
      1.8333333333333335,
      9.166666666666666,
    ]),
    list([
      25,
      0,
      0,
      1,
      1.5,
      1.8333333333333335,
      9.166666666666666,
    ]),
    list([
      26,
      0,
      0,
      1,
      1.8333333333333335,
      1.8333333333333335,
      9.166666666666666,
    ]),
    list([
      27,
      0,
      1,
      1,
      1.1666666666666667,
      0.16666666666666666,
      5.833333333333333,
    ]),
    list([
      28,
      0,
      1,
      1,
      1.5,
      0.16666666666666666,
      5.833333333333333,
    ]),
    list([
      29,
      0,
      1,
      1,
      1.8333333333333335,
      0.16666666666666666,
      5.833333333333333,
    ]),
    list([
      30,
      0,
      1,
      1,
      1.1666666666666667,
      0.5,
      5.833333333333333,
    ]),
    list([
      31,
      0,
      1,
      1,
      1.5,
      0.5,
      5.833333333333333,
    ]),
    list([
      32,
      0,
      1,
      1,
      1.8333333333333335,
      0.5,
      5.833333333333333,
    ]),
    list([
      33,
      0,
      1,
      1,
      1.1666666666666667,
      0.8333333333333333,
      5.833333333333333,
    ]),
    list([
      34,
      0,
      1,
      1,
      1.5,
      0.8333333333333333,
      5.833333333333333,
    ]),
    list([
      35,
      0,
      1,
      1,
      1.8333333333333335,
      0.8333333333333333,
      5.833333333333333,
    ]),
    list([
      36,
      0,
      1,
      1,
      1.1666666666666667,
      0.16666666666666666,
      7.5,
    ]),
    list([
      37,
      0,
      1,
      1,
      1.5,
      0.16666666666666666,
      7.5,
    ]),
    list([
      38,
      0,
      1,
      1,
      1.8333333333333335,
      0.16666666666666666,
      7.5,
    ]),
    list([
      39,
      0,
      1,
      1,
      1.1666666666666667,
      0.5,
      7.5,
    ]),
    list([
      40,
      0,
      1,
      1,
      1.5,
      0.5,
      7.5,
    ]),
    list([
      41,
      0,
      1,
      1,
      1.8333333333333335,
      0.5,
      7.5,
    ]),
    list([
      42,
      0,
      1,
      1,
      1.1666666666666667,
      0.8333333333333333,
      7.5,
    ]),
    list([
      43,
      0,
      1,
      1,
      1.5,
      0.8333333333333333,
      7.5,
    ]),
    list([
      44,
      0,
      1,
      1,
      1.8333333333333335,
      0.8333333333333333,
      7.5,
    ]),
    list([
      45,
      0,
      1,
      1,
      1.1666666666666667,
      0.16666666666666666,
      9.166666666666666,
    ]),
    list([
      46,
      0,
      1,
      1,
      1.5,
      0.16666666666666666,
      9.166666666666666,
    ]),
    list([
      47,
      0,
      1,
      1,
      1.8333333333333335,
      0.16666666666666666,
      9.166666666666666,
    ]),
    list([
      48,
      0,
      1,
      1,
      1.1666666666666667,
      0.5,
      9.166666666666666,
    ]),
    list([
      49,
      0,
      1,
      1,
      1.5,
      0.5,
      9.166666666666666,
    ]),
    list([
      50,
      0,
      1,
      1,
      1.8333333333333335,
      0.5,
      9.166666666666666,
    ]),
    list([
      51,
      0,
      1,
      1,
      1.1666666666666667,
      0.8333333333333333,
      9.166666666666666,
    ]),
    list([
      52,
      0,
      1,
      1,
      1.5,
      0.8333333333333333,
      9.166666666666666,
    ]),
    list([
      53,
      0,
      1,
      1,
      1.8333333333333335,
      0.8333333333333333,
      9.166666666666666,
    ]),
  ])
# ---
# name: test_lrcparticledata_to_prp_divisions_defaults[True]
  list([
    list([
      0,
      0,
      0,
      1,
      1.1666666666666667,
      1.1666666666666667,
      0.16666666666666666,
    ]),
    list([
      1,
      0,
      0,
      1,
      1.5,
      1.1666666666666667,
      0.16666666666666666,
    ]),
    list([
      2,
      0,
      0,
      1,
      1.8333333333333335,
      1.1666666666666667,
      0.16666666666666666,
    ]),
    list([
      3,
      0,
      0,
      1,
      1.1666666666666667,
      1.5,
      0.16666666666666666,
    ]),
    list([
      4,
      0,
      0,
      1,
      1.5,
      1.5,
      0.16666666666666666,
    ]),
    list([
      5,
      0,
      0,
      1,
      1.8333333333333335,
      1.5,
      0.16666666666666666,
    ]),
    list([
      6,
      0,
      0,
      1,
      1.1666666666666667,
      1.8333333333333335,
      0.16666666666666666,
    ]),
    list([
      7,
      0,
      0,
      1,
      1.5,
      1.8333333333333335,
      0.16666666666666666,
    ]),
    list([
      8,
      0,
      0,
      1,
      1.8333333333333335,
      1.8333333333333335,
      0.16666666666666666,
    ]),
    list([
      9,
      0,
      0,
      1,
      1.1666666666666667,
      1.1666666666666667,
      0.5,
    ]),
    list([
      10,
      0,
      0,
      1,
      1.5,
      1.1666666666666667,
      0.5,
    ]),
    list([
      11,
      0,
      0,
      1,
      1.8333333333333335,
      1.1666666666666667,
      0.5,
    ]),
    list([
      12,
      0,
      0,
      1,
      1.1666666666666667,
      1.5,
      0.5,
    ]),
    list([
      13,
      0,
      0,
      1,
      1.5,
      1.5,
      0.5,
    ]),
    list([
      14,
      0,
      0,
      1,
      1.8333333333333335,
      1.5,
      0.5,
    ]),
    list([
      15,
      0,
      0,
      1,
      1.1666666666666667,
      1.8333333333333335,
      0.5,
    ]),
    list([
      16,
      0,
      0,
      1,
      1.5,
      1.8333333333333335,
      0.5,
    ]),
    list([
      17,
      0,
      0,
      1,
      1.8333333333333335,
      1.8333333333333335,
      0.5,
    ]),
    list([
      18,
      0,
      0,
      1,
      1.1666666666666667,
      1.1666666666666667,
      0.8333333333333333,
    ]),
    list([
      19,
      0,
      0,
      1,
      1.5,
      1.1666666666666667,
      0.8333333333333333,
    ]),
    list([
      20,
      0,
      0,
      1,
      1.8333333333333335,
      1.1666666666666667,
      0.8333333333333333,
    ]),
    list([
      21,
      0,
      0,
      1,
      1.1666666666666667,
      1.5,
      0.8333333333333333,
    ]),
    list([
      22,
      0,
      0,
      1,
      1.5,
      1.5,
      0.8333333333333333,
    ]),
    list([
      23,
      0,
      0,
      1,
      1.8333333333333335,
      1.5,
      0.8333333333333333,
    ]),
    list([
      24,
      0,
      0,
      1,
      1.1666666666666667,
      1.8333333333333335,
      0.8333333333333333,
    ]),
    list([
      25,
      0,
      0,
      1,
      1.5,
      1.8333333333333335,
      0.8333333333333333,
    ]),
    list([
      26,
      0,
      0,
      1,
      1.8333333333333335,
      1.8333333333333335,
      0.8333333333333333,
    ]),
    list([
      27,
      0,
      1,
      1,
      1.1666666666666667,
      0.16666666666666666,
      0.16666666666666666,
    ]),
    list([
      28,
      0,
      1,
      1,
      1.5,
      0.16666666666666666,
      0.16666666666666666,
    ]),
    list([
      29,
      0,
      1,
      1,
      1.8333333333333335,
      0.16666666666666666,
      0.16666666666666666,
    ]),
    list([
      30,
      0,
      1,
      1,
      1.1666666666666667,
      0.5,
      0.16666666666666666,
    ]),
    list([
      31,
      0,
      1,
      1,
      1.5,
      0.5,
      0.16666666666666666,
    ]),
    list([
      32,
      0,
      1,
      1,
      1.8333333333333335,
      0.5,
      0.16666666666666666,
    ]),
    list([
      33,
      0,
      1,
      1,
      1.1666666666666667,
      0.8333333333333333,
      0.16666666666666666,
    ]),
    list([
      34,
      0,
      1,
      1,
      1.5,
      0.8333333333333333,
      0.16666666666666666,
    ]),
    list([
      35,
      0,
      1,
      1,
      1.8333333333333335,
      0.8333333333333333,
      0.16666666666666666,
    ]),
    list([
      36,
      0,
      1,
      1,
      1.1666666666666667,
      0.16666666666666666,
      0.5,
    ]),
    list([
      37,
      0,
      1,
      1,
      1.5,
      0.16666666666666666,
      0.5,
    ]),
    list([
      38,
      0,
      1,
      1,
      1.8333333333333335,
      0.16666666666666666,
      0.5,
    ]),
    list([
      39,
      0,
      1,
      1,
      1.1666666666666667,
      0.5,
      0.5,
    ]),
    list([
      40,
      0,
      1,
      1,
      1.5,
      0.5,
      0.5,
    ]),
    list([
      41,
      0,
      1,
      1,
      1.8333333333333335,
      0.5,
      0.5,
    ]),
    list([
      42,
      0,
      1,
      1,
      1.1666666666666667,
      0.8333333333333333,
      0.5,
    ]),
    list([
      43,
      0,
      1,
      1,
      1.5,
      0.8333333333333333,
      0.5,
    ]),
    list([
      44,
      0,
      1,
      1,
      1.8333333333333335,
      0.8333333333333333,
      0.5,
    ]),
    list([
      45,
      0,
      1,
      1,
      1.1666666666666667,
      0.16666666666666666,
      0.8333333333333333,
    ]),
    list([
      46,
      0,
      1,
      1,
      1.5,
      0.16666666666666666,
      0.8333333333333333,
    ]),
    list([
      47,
      0,
      1,
      1,
      1.8333333333333335,
      0.16666666666666666,
      0.8333333333333333,
    ]),
    list([
      48,
      0,
      1,
      1,
      1.1666666666666667,
      0.5,
      0.8333333333333333,
    ]),
    list([
      49,
      0,
      1,
      1,
      1.5,
      0.5,
      0.8333333333333333,
    ]),
    list([
      50,
      0,
      1,
      1,
      1.8333333333333335,
      0.5,
      0.8333333333333333,
    ]),
    list([
      51,
      0,
      1,
      1,
      1.1666666666666667,
      0.8333333333333333,
      0.8333333333333333,
    ]),
    list([
      52,
      0,
      1,
      1,
      1.5,
      0.8333333333333333,
      0.8333333333333333,
    ]),
    list([
      53,
      0,
      1,
      1,
      1.8333333333333335,
      0.8333333333333333,
      0.8333333333333333,
    ]),
  ])
# ---
